From 2c8e8755b2b2d7313bc69f103063eb38d64c040a Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 8 Apr 2026 14:25:35 +0800
Subject: [PATCH 2/2] Revert "aplay: reorganize format handling in
 begin_wave()"

Upstream-Status: Pending

This reverts commit e78583ab7cde7a2ccbb56339dd51608fb78c5395.
---
 aplay/aplay.c | 32 ++++++++++++++++++++------------
 1 file changed, 20 insertions(+), 12 deletions(-)

diff --git a/aplay/aplay.c b/aplay/aplay.c
index 3dc599c..8356dff 100644
--- a/aplay/aplay.c
+++ b/aplay/aplay.c
@@ -2716,7 +2716,7 @@ static void begin_wave(int fd, size_t cnt)
 	WaveHeader h;
 	WaveFmtBody f;
 	WaveChunkHeader cf, cd;
-	int width, physical_width;
+	int bits;
 	uint32_t tmp;
 	uint16_t tmp2;
 
@@ -2724,21 +2724,23 @@ static void begin_wave(int fd, size_t cnt)
 	if (cnt == (size_t)-2)
 		cnt = 0x7fffff00;
 
-	width = snd_pcm_format_physical_width(hwparams.format);
-	physical_width = snd_pcm_format_width(hwparams.format);
-
-	if (width < 0 || physical_width < 0)
-		goto _format;
-
+	bits = 8;
 	switch ((unsigned long) hwparams.format) {
 	case SND_PCM_FORMAT_U8:
+		bits = 8;
+		break;
 	case SND_PCM_FORMAT_S16_LE:
-	case SND_PCM_FORMAT_S24_LE: /* S24_LE is 24 bits stored in 32 bit width with 8 bit padding */
+		bits = 16;
+		break;
 	case SND_PCM_FORMAT_S32_LE:
-	case SND_PCM_FORMAT_FLOAT_LE:
+        case SND_PCM_FORMAT_FLOAT_LE:
+		bits = 32;
+		break;
+	case SND_PCM_FORMAT_S24_LE:
 	case SND_PCM_FORMAT_S24_3LE:
+		bits = 24;
+		break;
 	default:
-_format:
 		error(_("Wave doesn't support %s format..."), snd_pcm_format_name(hwparams.format));
 		prg_exit(EXIT_FAILURE);
 	}
@@ -2756,11 +2758,17 @@ _format:
                 f.format = LE_SHORT(WAV_FMT_PCM);
 	f.channels = LE_SHORT(hwparams.channels);
 	f.sample_fq = LE_INT(hwparams.rate);
-	tmp2 = hwparams.channels * physical_width / 8;
+#if 0
+	tmp2 = (samplesize == 8) ? 1 : 2;
+	f.byte_p_spl = LE_SHORT(tmp2);
+	tmp = dsp_speed * hwparams.channels * (uint32_t) tmp2;
+#else
+	tmp2 = hwparams.channels * snd_pcm_format_physical_width(hwparams.format) / 8;
 	f.byte_p_spl = LE_SHORT(tmp2);
 	tmp = (uint32_t) tmp2 * hwparams.rate;
+#endif
 	f.byte_p_sec = LE_INT(tmp);
-	f.bit_p_spl = LE_SHORT(width);
+	f.bit_p_spl = LE_SHORT(bits);
 
 	cd.type = WAV_DATA;
 	cd.length = LE_INT(cnt);
-- 
2.34.1

